docs/build/guides/archival/extend-persistent-entry-js.mdx

```diff
@@ -0,0 +1,179 @@
+    durability: StellarSdk.xdr.ContractDataDurability.persistent(),
+  });
+
+  // Create footprint for the entry
+  const footprint = new StellarSdk.xdr.LedgerFootprint({
+    readOnly: [],
+    readWrite: [persistentEntry],
+  });
+
+  // build and submit the transaction
+  const transaction = new StellarSdk.TransactionBuilder(account, {
+    fee,
+    networkPassphrase: StellarSdk.Networks.TESTNET,
+  })
+    .addOperation(
+      StellarSdk.Operation.extendFootprintTtl({
+        extendTo: 500_000,
+        footprint: footprint,
+      }),
+    )
+    .setTimeout(30)
+    .build();
+
+  transaction.sign(sourceKeypair);
+  return await server.sendTransaction(transaction);
+}
+```
+
+### Extending the contract code
+
+```javascript
+const StellarSdk = require("@stellar/stellar-sdk");
+const { Server } = require("@stellar/stellar-sdk/rpc");
+
+async function extendContractCodeTTL(contractId, sourceKeypair) {
+  const server = new Server("https://soroban-testnet.stellar.org");
+  const account = await server.getAccount(sourceKeypair.publicKey());
+  const fee = "200100";
+
+  // Create the contract code identifier
+  const contractHash = new StellarSdk.xdr.LedgerKey.contractCode({
+    hash: StellarSdk.Address.fromString(contractId).toScAddress(),
+  });
+
+  // Create footprint for contract code
+  const footprint = new StellarSdk.xdr.LedgerFootprint({
+    readOnly: [],
+    readWrite: [contractHash],
+  });
+
+  // build and submit the transaction
+  const transaction = new StellarSdk.TransactionBuilder(account, {
+    fee,
+    networkPassphrase: StellarSdk.Networks.TESTNET,
+  })
+    .addOperation(
+      StellarSdk.Operation.extendFootprintTtl({
+        extendTo: 500_000,
+        footprint: footprint,
+      }),
+    )
+    .setTimeout(30)
+    .build();
+
+  transaction.sign(sourceKeypair);
+  return await server.sendTransaction(transaction);
+}
+```
+
+### Extending both the contract and a persistent entry
+
+```javascript
+const StellarSdk = require("@stellar/stellar-sdk");
+const { Server } = require("@stellar/stellar-sdk/rpc");
+
+async function extendContractAndPersistentEntry(
+  contractId,
+  storageKey,
+  sourceKeypair,
+) {
+  const server = new Server("https://soroban-testnet.stellar.org");
+  const account = await server.getAccount(sourceKeypair.publicKey());
+  const fee = "200100";
+
+  // Create both identifiers
+  const contractHash = new StellarSdk.xdr.LedgerKey.contractCode({
+    hash: StellarSdk.Address.fromString(contractId).toScAddress(),
+  });
+
+  const persistentEntry = new StellarSdk.xdr.LedgerKey.contractData({
+    contract: StellarSdk.Address.fromString(contractId).toScAddress(),
+    key: StellarSdk.xdr.ScVal.scvSymbol(storageKey),
+    durability: StellarSdk.xdr.ContractDataDurability.persistent(),
+  });
+
+  // Create footprint with both identifiers
+  const footprint = new StellarSdk.xdr.LedgerFootprint({
+    readOnly: [],
+    readWrite: [contractHash, persistentEntry],
```

Comment:
This needs to be executed as 2 separate operations

https://developers.stellar.org/docs/learn/encyclopedia/storage/state-archival#extendfootprintttlop

> ExtendFootprintTTLOp is a Soroban operation that will extend the live until ledger of the entries specified in the read-only set of the footprint. The read-write set must be empty.